2021 Update (#1215)
carpentries/instructor-training: _episodes/02-practice-learning.md

in her studies of
[how nurses progress from novice to expert](http://journals.sagepub.com/doi/10.1177/0270467604265061)
([see also books by Benner](https://www.worldcat.org/search?q=au%3ABenner%2C+Patricia+E.&qt=hot_author)). This work indicates that
through practice and formal instruction, learners acquire skills and advance through distinct stages. In simplified form, three stages of this model are:

![Three people, labeled from left to right as "Novice", "Competent Practitioner", and "Expert". Underneath the people,
an arrow labelled "Experience level" points from left to right. Underneath the figure labelled "Novice" a quote says "I'm not sure what questions to ask." The Competent Practitioner says "I'm pretty confident, but I still look stuff up a lot!" The Expert says "I've been doing this on a daily basis for years!"](../fig/skill-level.svg)

*   *Novice*: someone who does not know what they do not know, i.e.,
    they do not yet know what the key ideas in the domain are or how they relate.

>     Novices may have difficulty formulating questions, or may ask questions that seem irrelevant or off-topic
>     as they reach for anything they know to connect.

Reviewer: ```suggestion
    Novices may have difficulty formulating questions, or may ask questions that seem irrelevant or off-topic
    as they rely on prior knowledge, without knowing what is/not related yet.
```

"reach for anything they know how to connect was vague, but i think this captures?

    > Example: An *expert* in a Carpentries workshop may have experience writing and running shell scripts and, when
    presented with a problem, immediately sees how these skills can be used to solve the problem.

Note that, **while you may not think of yourself as a computational "expert", in the context above that term may accurately describe
your skill level** in some situations. We will come back to the expertise of the instructor and its impact -- positive and negative -- 
on teaching, in the next episode. 
For now, we are primarily concerned with novices,
as this tends to characterize The Carpentries audience.

It is common to think of a novice as a sort of an "empty vessel" into which knowledge can be "poured." Unfortunately, this analogy has
limited usefulness and is wrong in ways that generate dangerous misconceptions. In our next section, we will briefly explore 
the nature of "knowledge" through a concept
that helps us differentiate between novices and competent practitioners in a more useful and visual way. This, in turn, will have implications
for how we teach.

> ## "All Models are Wrong, but Some are Useful"

Reviewer 2: This title seems a little odd to me, we don't really talk about how all models are wrong... Maybe something simpler like "Building Mental Models"?

> hands on, practical, and useful. This clearly conveys the purpose of our organization: to support computational skill development
> among working practitioners who need the right tools and practices to be effective day to day.
{: .callout}

A mental model may be represented as a collection of concepts and facts, connected by relationships. 
The mental model of an expert in any given subject will be far larger and more complex than that of a novice, including both more concepts 
and more detailed and numerous relationships. However, **both may be perfectly useful** in certain contexts. 

Returning to our example levels of skill development: 
*     A *novice* has a minimal mental model of surface features of the domain. Inaccuracies may interfere with adding new information.
Predictions are likely to borrow heavily from mental models of other domains
which seem superficially similar.
*     A *competent practitioner* has a mental model that is useful for everyday purposes. Most new information
they are likely to encounter will fit well with their existing model. Even though many potential elements of their mental model may

> still be missing or wrong, predictions about their area of work are usually accurate.

Reviewer 2: Is there a reason we don't describe the expert mental model briefly here? It seems odd to include it in the diagram but not mention it.

Reviewer 2: I realize that we get to it later but I think it helps to define it here as well.

    > Example: An *expert* in a Carpentries workshop may have experience writing and running shell scripts and, when
    presented with a problem, immediately sees how these skills can be used to solve the problem.

Note that, while you may not think of yourself as a computational "expert," in the context above that term may accurately describe
your skill level. We will come back to the expertise of the instructor and its impact -- positive and negative -- on teaching, in the next episode. 
For now, we are primarily concerned with novices,

> as this tends to characterize The Carpentries audience.

Reviewer: ```suggestion
as this is The Carpentries target audience.
```

I vote for slighly stronger here in that this is the vision and that it's not a coincidence but we sometimes get others.  Carpentries materials are designed for novices and we have strategies to accomodate others to varying degrees.

Author: ```suggestion
as this is The Carpentries' primary target audience.
```
Agree; adding "primary" to be inclusive (and some lessons are more novice than others)

It is common to think of a novice as a sort of an "empty vessel" into which knowledge can be "poured." Unfortunately, this analogy has
limited usefulness and is wrong in ways that generate dangerous misconceptions. In our next section, we will briefly explore the nature of "knowledge" through a concept
that helps us differentiate between novices and competent practitioners in a more useful and visual way. This, in turn, will have implications

> for how we teach.

Reviewer: it might be good to be more specific than "useful" here because we truly care about what is actionable for teaching strategies, not other types of "useful" an instructor could have.

"Knowledge" is hard to describe. Understanding is never a mirror of reality, even for an expert; rather, it is 
an internal representation based on our experience with a subject. 
This internal representation is often described as a **mental model**. A mental model
allows us to extrapolate, or make predictions beyond and between the narrow limits of experience and memory, filling in 
gaps to the point that things "make sense." 

As we learn, our mental model evolves to become more useful. A useful model makes reasonable predictions and fits well within
the range of things
we are likely to encounter, or at least does not break down entirely as new concepts are added. A driver of a gasoline powered car may 
do just fine with a mental model that includes relationships between fuel, an engine, and a car battery;
a broken alternator may not be something they could predict, but learning that another part mediates the relationship 
between the engine and the battery adds to the model without the driver having to add any other concepts or un-learn mis-conceived relationships. 
Alternatively, a child who attributes sentient will to a vehicle will have to work much harder to make sense 

> of a broken alternator.

Reviewer: this is hard to parse for me at a couple of levels:
- I think a little bit of grammatical revision could make this example more readable
- I'm not sure what the takeaway is supposed to be here and what the most salient parts of this example are. 

I appreciate the goal of not using something programming related that relies on a. more sophisticated mental model than what we teach, because it can leave out some trainees.  At the same time, something a little closer can also start to get trainees to start thinking in terms of this.  Maybe we can source from the trainer community a few examples here?  I think this is a place where people use their own examples often anyway.

Reviewer: this could be my relative comfort with concept maps, but i also feel like using this example here to show a very simple concept map -- not trying to teach them how to write their own yet, jsut *using* one to externalize the mental model -- could be helpful here

Author: I'm going to interpret this comment to indicate that the restructuring mostly works, but the example and the mechanics of connection between mental models and concept maps may need some work. On this basis, I plan to let this sit for a while while I work on other sections and think about these comments. :) Please let me know if I have that wrong!

I struggled to come up with an example that really did what I wanted it to do, which is to present a sort of mechanical causality that was 1) generally relatable, 2) simple and clear both with and without a missing element, and 3) included a misconception that would obviously interfere with understanding the missing element. Other examples I have seen Trainers use do a great job of illustrating concept mapping or mental models but don't check all these boxes. However, there are certainly other examples that may work better, including some that are more proximal, like relationships between hardware components or maybe the difference between streaming services and DVDs... things that break (or don't) when the internet fails? Will keep thinking and consider crowd-sourcing!

Reviewer 3: As an aside, I think my experiment with task decomposition instead of concept maps achieved my objectives. Not sure if it's worth bringing it in as an alternate during this revision cycle though.

Author: I think the formerly Blooms-based activity in Episode 17 now resembles this task decomposition activity.

identify how they are connected. It can be quite difficult to organize knowledge in this way! However, the process of forcing abstract knowledge into a visual 
format can often reveal connections you may not have been aware of, or illuminate gaps. Especially where analogies are not available, concept mapping can help 
you to make your mental model of a concept more clear to yourself or others.

![Four words inside circles, with labeled arrows connecting them. "Car" is at the top, with an arrow pointing to "engine" labeled as "is powered by." An arrow connects "engine" to "fuel," at left, labeled "requires energy from." Another arrow connects "engine" to "battery," at right, labeled "charges." An arrow connects "battery" back to "car," labeled "is needed to start."](../fig/Cmap-Car.svg)

> ## Mapping a Mental Model
>
> 1. On a piece of paper, draw a concept map of the same concept you discussed in the last activity, but this time without the analogy. What are 3-4 core 
> concepts involved? How are those concepts related?
>  
> 2. In the Etherpad, write some notes on this process. Was it frustrating? Do you think it would be a useful exercise prior to teaching about your topic?

> > What would be necessary for a learner to complete such an exercise?

Reviewer: what does such an exercise reference?

> ## Modeling Novice Mental Models
>
> Take 10 minutes to create a multiple choice question related to a lesson you intend to teach.
> 1. Think about the topic of the lesson. What relevant misconceptions might a novice learner bring to the classroom? 
> In what ways might they misunderstand or inappropriately connect concepts?

> > 2. Try to target one or more misconceptions using a multiple choice question. How many diagnostic distractors can you create?

Reviewer: ```suggestion
> 2. Try to target one or more misconceptions using a multiple choice question. How many plausible distractors can you create?
```
 to parallel what's abvoe
